File: app/src/main/java/com/caramelheaven/lennach/di/application/AppComponent.java
package com.caramelheaven.lennach.di.application;

import com.caramelheaven.lennach.di.main.MainComponent;
import com.caramelheaven.lennach.di.main.MainModule;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = {NetworkModule.class, AppModule.class})
public interface AppComponent {
    MainComponent plusMainComponent(MainModule mainModule);
}

File: app/src/main/java/com/caramelheaven/lennach/di/application/AppComponent.kt
package com.caramelheaven.lennach.di.application

import dagger.Component
import javax.inject.Singleton

/**
 * Created by CaramelHeaven on 16:48, 03/02/2019.
 */
@Singleton
@Component(modules = [AppModule::class])
interface AppComponent {
    //fun plusMainComponent(mainModule: MainModule): MainComponent
}
